File: SungJae Hong/2022-09-13-Best-Time-to-Buy-and-Sell-Stock.py
from typing import List

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        #left = buy, right = sell
        l, r = 0, 1
        #first intialize maximum profit to 0.
        max_profit = 0
        while r < len(prices):
            if prices[l] < prices[r]:
                profit = prices[r] - prices[l]
                max_profit = max(max_profit, profit)
            else:
                l = r
            r += 1
        return max_profit

File: DonRyu/2022-09-13-Best-Time-To-Buy-and-sell-stock.js

const maxProfit = (prices) => {

  let buy = prices[0]
  let profit = 0;

  for (let i = 1; i < prices.length; i++) {
    if (buy > prices[i]) {
      buy = prices[i]
    } else {
      profit = Math.max(prices[i] - buy, profit)
    }
  }
  return profit

}


console.log((maxProfit([7, 1, 5, 3, 6, 4])))
